packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -546,31 +585,36 @@ class _TextSelectionHandleOverlayState extends State<_TextSelectionHandleOverlay
         break;
     }
 
+    point = widget.renderObject.clampToViewport(point);
```

Comment:
For my own understanding: Why is this necessary?

Reply:
This prevents the handles from drawing outside the box while they're fading out:

![before](https://user-images.githubusercontent.com/29863079/52866283-a2c40700-313e-11e9-916d-abcbc9075b95.gif)

Instead we get:

![after](https://user-images.githubusercontent.com/29863079/52866290-a5bef780-313e-11e9-8e5d-3c5c0041ce48.gif)

The second is more similar to the native behavior on Android, and I think it looks better than the first.